ui/core/proto_utils/equipped_item.ts

```diff
@@ -293,6 +385,14 @@ export class EquippedItem {
 		return !!this._item.randomSuffixOptions.length;
 	}
 
+	hasUpgradeOptions() {
+		const { scalingOptions } = this.item;
+		// Make sure to always exclude Challenge Mode scaling options as those are handled globally
+		// and offset these options by 1 due to items always having a base option.
+		delete scalingOptions[ItemLevelState.ChallengeMode];
+		return Object.keys(scalingOptions).length > 1;
+	}
```

Comment:
Redundant code with `database.ts`, you should be able to just call
```
return Database.getAvailableUpgrades(this.item).length > 1;
```
where the CM filter is already applied within `getAvailableUpgrades()`. Or alternatively, delete that code within `database.ts` and move it here as a static method on `EquippedItem`. However you decide to do it, the CM filtering should only happen in one place rather than three.